### DISCUSSION TITLE
Can EPOCH seperate electron's radiation from laser

### DISCUSSION THREAD

**Discussion by ZhixinFan-hub** [epochpic/epoch#732] [Ideas]
When conducting simulations on topics such as high-order harmonic generation, structural optics, or wakefield acceleration, the incident laser is often significantly reshaped and electrons can emit strong radiation. I am particularly interested in determining the contributions of electron radiation to these effects.

In each simulation, we can output key physical parameters of electrons, such as gamma, mass, position, etc. This data should theoretically allow us to calculate the electromagnetic fields generated by the electrons and differentiate these from the reshaped laser fields. This process could enhance our understanding of electron dynamics.

I am wondering if EPOCH has the capability to perform these functions, or if there are other software tools that can achieve this. Does anyone have insights or suggestions? :)

**Comment by Status-Mirror** [comment]
I've seen groups do something similar to this before. Using the [subset block](https://epochpic.github.io/documentation/input_deck/input_deck_subset.html), you can force a small number of cells to output the E and B fields on most time-steps, without generating too much data in SDF files. This gives you a time-series of the E and B fields in these cells, and from this, you can get a time-series of the Poynting flux. 

By performing a Fourier transform, you can filter the Poynting flux to figure out how much energy is being carried by waves within a certain frequency range. This would allow you to distinguish between laser-frequency radiaiton, and that caused in high-harmonic generation.

Check out Fig 4 of [Seaton's paper](https://pubs.aip.org/aip/pop/article/27/8/082704/108558) for an example of when this has been in ICF simulations, where EM waves at the laser frequency were filtered from those produced in laser-plasma-instabilities.

Hope this helps,
Stuart

**Reply by ZhixinFan-hub** [reply]
Hi Stuart

It's true that Fourier transformation can help us filter the laser.
But as I said before, there are two resources: distortion of incident laser and radiation from electrons, especially in high order harmonics generation process, which lead to two distinct HHG mechanisms. However, in EPOCH they are mixed, subset block will only output the mixture of them.
Once we know all electron's movement via EPOCH simulation, I wonder how can we calculate their radiation without the effect of distorted incident laser ;)

ZHIXIN

**Reply by TomGoffrey** [reply]
You could, from the particle position velocity and acceleration, calculate the instantaneous EM field due to that macro particle. This is covered in many standard EM texts. Whether or not this would answer your research question is not something we can help you with.

EPOCH solves Maxwell's equations for the total EM fields. There's no splitting of the fields into different parts.
